worldedit-core/src/main/java/com/sk89q/worldedit/command/UtilityCommands.java

```diff
@@ -510,30 +511,36 @@ public void remove(Actor actor, CommandContext args) throws WorldEditException,
     }
 
     @Command(
-        aliases = { "/calc", "/calculate", "/eval", "/evaluate", "/solve" },
-        usage = "<expression>",
-        desc = "Evaluate a mathematical expression"
+            aliases = { "/calc", "/calculate", "/eval", "/evaluate", "/solve" },
+            usage = "<expression>",
+            desc = "Evaluate a mathematical expression"
     )
     @CommandPermissions("worldedit.calc")
-    public void calc(Actor actor, @Text String input) throws CommandException {
-        try {
-            Expression expression = Expression.compile(input);
-            actor.print("= " + expression.evaluate());
-        } catch (EvaluationException e) {
-            actor.printError(String.format(
-                    "'%s' could not be parsed as a valid expression", input));
-        } catch (ExpressionException e) {
-            actor.printError(String.format(
-                    "'%s' could not be evaluated (error: %s)", input, e.getMessage()));
-        }
+    public void calc(final Actor actor, @Text final String input) throws CommandException {
+        // executing as a thread should keep the server from crashing when using for loops.
```

Comment:
This isn't a proper solution, and may still cause crashes if enough commands are executed. A better solution would be to timeout calculations after a certain time.